core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -189,6 +189,18 @@ private[spark] class TaskSetManager(
     addPendingTask(i)
   }
 
+  {
+    // TaskSet got submitted by DAGScheduler may have some already completed
```

Comment:
shall we do it in `TaskSchedulerImpl.createTaskSetManager`? In general I'd avoid adding too much code to the constructor.

Reply:
Actually, I have the same thought. Let me try it.